pkg/scene/widget_align_system.go

```diff
@@ -0,0 +1,250 @@
+package scene
+
+import (
+	"github.com/reearth/reearth-backend/pkg/id"
+)
+
+// WidgetAlignSystem is the layout structure of any enabled widgets that will be displayed over the scene.
+type WidgetAlignSystem struct {
+	inner WidgetZone
+	outer WidgetZone
+}
+
+// WidgetZone is the structure of each layer (inner and outer) of the align system.
+type WidgetZone struct {
+	left   WidgetSection
+	center WidgetSection
+	right  WidgetSection
+}
+
+// WidgetSection is the structure of each section of the align system.
+type WidgetSection struct {
+	top    WidgetArea
+	middle WidgetArea
+	bottom WidgetArea
+}
+
+// WidgetArea has the widgets and alignment information found in each part area of a section.
+type WidgetArea struct {
+	widgetIds []*id.WidgetID
```

Comment:
```suggestion
	widgetIds []id.WidgetID
```

IDs can only be reference types if you want to make it clear that they can be nil. Most of the time, references are not needed in a ID slice.

Returned value type of methods also need to be fixed